runtime/marshal_jsonpb.go

```diff
@@ -151,7 +151,15 @@ func (d DecoderWrapper) Decode(v interface{}) error {
 
 // NewEncoder returns an Encoder which writes JSON stream into "w".
 func (j *JSONPb) NewEncoder(w io.Writer) Encoder {
-	return EncoderFunc(func(v interface{}) error { return j.marshalTo(w, v) })
+	return EncoderFunc(func(v interface{}) error {
+		if err := j.marshalTo(w, v); err != nil {
+			return err
+		}
+		// mimic json.Encoder by adding a newline (makes output
+		// easier to read when it contains multiple encoded items)
+		_, err := w.Write(j.Delimiter())
```

Comment:
In `etcd` this may be written to different HTTP chunk, right? As far as I know, the protocol must guarantee that 1 chunk = 1 message (JSON).

I don't understand go, but I have the code that triggers that. I mean, sometimes newline is seen as separate HTTP chunk.

Reply:
Moreover. What also stops from gluing two messages in one HTTP chunk?

What I propose: is to document that every message is separated strictly by one newline. In that case, there is possible to write stable stream parsing that does not rely on HTTP chunk boundary.

Reply:
Could you please raise a separate issue detailing your concerns? I don't think this change covers the behaviour you're asking about.

Reply:
#1086